ftw/simplelayout/browser/leadimage.py

```diff
@@ -0,0 +1,50 @@
+from ftw.simplelayout.contents.interfaces import ITextBlock
+from ftw.simplelayout.handlers import unwrap_persistence
+from ftw.simplelayout.interfaces import IPageConfiguration
+from plone.app.uuid.utils import uuidToObject
+from plone.memoize import ram
+from Products.Five.browser import BrowserView
+import json
+import re
+import hashlib
+
+
+def _render_cachkey(method, self):
+    key = (str(self.context.modified().millis()),
```

Comment:
Since the cachekey is global, it should contain a distinct identifier.
I would add the method dottedname to the cachekey.